File: src/sports_management/modules/booking_module/lib/domain/usecases/create_booking_usecase.dart
import 'package:server_module/server_module.dart';

/// Gọi POST /booking/ để tạo lịch đặt sân
class CreateBookingUseCase {
  final BookingRepository _bookingRepository;

  CreateBookingUseCase(this._bookingRepository);

  /// [bookingDate] dạng "yyyy-MM-dd", e.g. "2026-06-15"
  /// [startMinutes] và [endMinutes] là số phút từ 00:00, e.g. 420 = 07:00
  Future<BaseResponse<BookingEntity>> call({
    required String courtId,
    required String bookingDate,
    required int startMinutes,
    required int endMinutes,
    required double totalPrice,
    String? userId,
    String? guestName,
    String? guestPhone,
  }) async {
    return _bookingRepository.createBooking({
      'courtId': courtId,
      'bookingDate': bookingDate,
      'startMinutes': startMinutes,
      'endMinutes': endMinutes,
      'totalPrice': totalPrice,
      'userId': ?userId,
      'guestName': ?guestName,
      'guestPhone': ?guestPhone,
    });
  }
}

File: src/react-staff-admin/src/features/booking/domain/usecases/create_booking.usecase.ts
import { BookingRepository } from '../repositories/booking.repository';
import { Booking } from '../entities/booking.entity';

export class CreateBookingUseCase {
  constructor(private repository: BookingRepository) {}

  async execute(
    courtId: string,
    bookingDate: string,
    startMinutes: number,
    endMinutes: number,
    totalPrice: number,
    userId?: string
  ): Promise<Booking> {
    return this.repository.createBooking(courtId, bookingDate, startMinutes, endMinutes, totalPrice, userId);
  }
}
